applications/ShapeOptimizationApplication/python_scripts/analyzer_internal.py

```diff
@@ -28,8 +28,8 @@ def __init__( self, specified_responses, model_part_controller ):
 
     # --------------------------------------------------------------------------
     def InitializeBeforeOptimizationLoop( self ):
-        for function in self.response_functions.values():
-            function.Initialize()
+        for _, response in self.response_functions:

```

Comment:
why not looping over `.values` if you are not using the key?

Reply:
Note that response_functions is a list now. But actually, this doesn't seem very clever anymore. I changed it back , which also resolves your comments below.